client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -1,36 +1,163 @@
+			} );
+			setAttributes( { paymentId: id } );
+		} catch ( err ) {
+			console.error( err );
+		}
+	};
+
+	handleEmailChange = event => {
+		this.props.setAttributes( { email: event.target.value } );
+	};
+
+	handleDescriptionChange = event => {
+		this.props.setAttributes( { description: event.target.value } );
+	};
+
+	handlePriceChange = event => {
+		const price = parseFloat( event.target.value );
+		if ( ! isNaN( price ) ) {
+			this.props.setAttributes( { price } );
 		} else {
-			this.props.setAttributes( { paymentId: undefined } );
+			this.props.setAttributes( { price: undefined } );
 		}
 	};
 
+	handleTitleChange = event => {
+		this.props.setAttributes( { title: event.target.value } );
+	};
+
 	render() {
 		const { attributes, instanceId } = this.props;
-		const inputId = `payment-id-${ instanceId }`;
+		const { title, /*currency,*/ description, price, /*multiple,*/ email } = attributes;
+		const baseId = `simplepayments-${ instanceId }`;
+		const titleId = `${ baseId }__title`;
+		const descriptionId = `${ baseId }__description`;
+		const currencyId = `${ baseId }__currency`;
+		const priceId = `${ baseId }__price`;
+		const multipleId = `${ baseId }__multiple`;
+		const emailId = `${ baseId }__email`;
+
 		return (
 			<Fragment>
-				<label htmlFor={ inputId }>Simple Payment ID</label>
-				<input
-					id={ inputId }
-					onChange={ this.handleChange }
-					type="number"
-					value={ attributes.paymentId || '' }
-				/>
+				<div>
+					<label htmlFor={ titleId }>Title</label>
+					<input id={ titleId } onChange={ this.handleTitleChange } type="text" value={ title } />
+				</div>
+				<div>
+					<label htmlFor={ descriptionId }>Description</label>
+					<textarea
+						id={ descriptionId }
+						onChange={ this.handleDescriptionChange }
+						value={ description }
+					/>
+				</div>
+				<div>
+					<label htmlFor={ currencyId }>Currency</label>
+					<input id={ currencyId } type="text" value="USD" />
+				</div>
+				<div>
+					<label htmlFor={ priceId }>Price</label>
+					<input
+						id={ priceId }
+						min={ 0.01 }
+						onChange={ this.handlePriceChange }
+						step={ 0.01 }
+						type="number"
+						value={ price ? price.toFixed( 2 ) : '' }
+					/>
+				</div>
+				<div>
+					<label htmlFor={ multipleId }>
+						Allow multiple
+						<input disabled id={ multipleId } type="checkbox" />
+					</label>
+				</div>
+				<div>
+					<label htmlFor={ emailId }>Email</label>
+					<input id={ emailId } onChange={ this.handleEmailChange } type="email" value={ email } />
+				</div>
 			</Fragment>
 		);
 	}
 }
 
 export default withInstanceId( Edit );
+
+function attributesToPost( attributes ) {
+	const { title, description, price, email } = attributes;
+
+	return {
+		title,
+		status: 'publish',
+		content: description,
+		featured_media: '',
+		// @FIXME: meta isn't saving. Tried { spay_price: … } simple object literal and it didn't work either :(
```

Comment:
We need to use `register_meta` to whitelist these meta fields in the API for reading+writing (working on it).

https://developer.wordpress.org/rest-api/extending-the-rest-api/modifying-responses/